specs/AppNotifications/AppNotificationContentSpec/AppNotificationBuilder-spec.md

```diff
@@ -0,0 +1,1178 @@
+        AppNotificationContent SetScenario(AppNotificationScenario value);
+
+        // Adds text to the AppNotification.
+        AppNotificationContent AddText(String text);
+        AppNotificationContent AddText(String text, AppNotificationTextProperties properties);
+
+        AppNotificationContent SetAttributionText(String text);
+        AppNotificationContent SetAttributionText(String text, String language);
+
+        // Sets the full-width inline-image that appears when you expand the AppNotification
+        AppNotificationContent SetInlineImage(Windows.Foundation.Uri imageUri);
+        [default_overload]
+        AppNotificationContent SetInlineImage(Windows.Foundation.Uri imageUri, ImageCrop imageCrop);
+        AppNotificationContent SetInlineImage(Windows.Foundation.Uri imageUri, String alternateText);
+        AppNotificationContent SetInlineImage(Windows.Foundation.Uri imageUri, String alternateText, ImageCrop imagecrop);
+
+        // Sets the image that is left-aligned with notification text
+        AppNotificationContent SetAppLogoOverride(Windows.Foundation.Uri imageUri);
+        AppNotificationContent SetAppLogoOverride(Windows.Foundation.Uri imageUri, String alternateText);
+        AppNotificationContent SetAppLogoOverride(Windows.Foundation.Uri imageUri, ImageCrop imageCrop);
+        AppNotificationContent SetAppLogoOverride(Windows.Foundation.Uri imageUri, String alternateText, ImageCrop imageCrop);
+
+        // Sets the image that displays within the banner of the AppNotification.
+        AppNotificationContent SetHeroImage(Windows.Foundation.Uri imageUri);
+        AppNotificationContent SetHeroImage(Windows.Foundation.Uri imageUri, String alternateText);
+
+        // SetAudio
+        [default_overload]
+        AppNotificationContent SetAudio(Windows.Foundation.Uri audioUri);
+        AppNotificationContent SetAudio(AppNotficiationSoundEvent appNotficiationSoundEvent);
+
+        [default_overload]
+        AppNotificationContent SetAudio(Windows.Foundation.Uri audioUri, AppNotificationDuration duration);
+        AppNotificationContent SetAudio(AppNotficiationSoundEvent appNotficiationSoundEvent, AppNotificationDuration duration);
+
+        AppNotificationContent MuteAudio();
+
+        // Add an input textbox to retrieve user input.
+        AppNotificationContent AddTextBox(String id);
+        AppNotificationContent AddTextBox(String id, String placeHolderText, String title);
+
+        // Adds a button to the AppNotificationContent
+        AppNotificationContent AddButton(AppNotificationButton value);
+
+        // Add an input ComboBox to retrieve user input.
+        AppNotificationContent AddComboBox(AppNotificationComboBox value);
+
+        AppNotificationContent AddProgressBar(AppNotificationProgressBar value);
+
+        // Retrieves the notification XML content so that it can be used with a local
+        // AppNotification constructor.
+        String GetXml();
```

Comment:
Do we also want a `Clone()` method, so you can take somebody else's builder and modify the content? E.g. if you want to raise multiple similar toasts. (Alternatively, "CreateFromXml")

Reply:
Consider for the future.